e2e/tests/params_tx_test.go

```diff
@@ -0,0 +1,138 @@
+	s.once.Do(func() {
+		s.Cleanup(func() { s.resetAllModuleParamsToDefaults() })
+	})
+}
+
+// newTempTxJSONFile creates & returns a new temp file with the JSON representation of a tx
+// which contains a MsgUpdateParams to update **all module params** for each module & paramsMap
+// in the given moduleParamsMap. The returned file is intended for use with the `authz exec` CLI
+// subcommand: `poktrolld tx authz exec <tx_json_file>`.
+func (s *suite) newTempUpdateParamsTxJSONFile(moduleParams moduleParamsMap) *os.File {
+	s.Helper()
+
+	var anyMsgs []*types.Any
+
+	// Collect msgs to update all params (per msg) for each module.
+	// E.g., 3 modules with 2 params each will result in 3 MsgUpdateParams messages in one tx.
+	for moduleName, paramsMap := range moduleParams {
+		// Convert the params map to a MsgUpdateParams message.
+		msgUpdateParams := s.paramsMapToMsgUpdateParams(moduleName, paramsMap)
+
+		// Convert the MsgUpdateParams message to a pb.Any message.
+		anyMsg, err := types.NewAnyWithValue(msgUpdateParams)
+		require.NoError(s, err)
+
+		anyMsgs = append(anyMsgs, anyMsg)
+	}
+
+	return s.newTempTxJSONFile(anyMsgs)
+}
+
+// newTempUpdateParamTxJSONFile creates & returns a new temp file with the JSON representation of a tx
+// which contains a MsgUpdateParam to update params **individually** for each module & paramsMap in the
+// given moduleParamsMap. The returned file is intended for use with the `authz exec` CLI subcommand:
+// `poktrolld tx authz exec <tx_json_file>`.
+func (s *suite) newTempUpdateParamTxJSONFile(moduleParams moduleParamsMap) *os.File {
+	s.Helper()
+
+	var anyMsgs []*types.Any
+
+	// Collect msgs to update given params, one param per msg, for each module.
+	// E.g., 3 modules with 2 given params each will result in 6 MsgUpdateParam messages in one tx.
+	for moduleName, paramsMap := range moduleParams {
+		for _, param := range paramsMap {
+			// Convert the params map to a MsgUpdateParam message.
+			msgUpdateParam := s.newMsgUpdateParam(moduleName, param)
+
+			// Convert the MsgUpdateParams message to a pb.Any message.
+			anyMsg, err := types.NewAnyWithValue(msgUpdateParam)
+			require.NoError(s, err)
+
+			anyMsgs = append(anyMsgs, anyMsg)
+		}
+	}
+
+	return s.newTempTxJSONFile(anyMsgs)
+}
+
+// newTempTxJSONFile creates & returns a new temp file with the JSON representation
+// of a tx which contains the given pb.Any messages. The temp file is removed when
+// the test completes.
+func (s *suite) newTempTxJSONFile(anyMsgs []*types.Any) *os.File {
+	s.Helper()
+
+	// Construct a TxBody with the pb.Any message for serialization.
+	txBody := &tx.TxBody{
+		Messages: anyMsgs,
+	}
+
+	// Serialize txBody to JSON for interpolation into the tx JSON template.
+	txBodyJSON, err := s.cdc.MarshalJSON(txBody)
+	require.NoError(s, err)
+
+	// Create a temporary file to write the interpolated tx JSON.
+	tempFile, err := os.CreateTemp("", "exec.json")
+	require.NoError(s, err)
+
+	defer func(f *os.File) {
```

Comment:
Isn't it possible to just call `defer tempFile.Close()`?

Reply:
It is but once we turn on more linters, one should complain about the ignored error return from `#Close()`.